### PR TITLE
Redirect taxonomy references to new repository

### LINKED ISSUE #1832: Extract taxonomy related CLI commands from the Fybrik main repository
Currently, taxonomy CLI commands are part of the main Fybrik repo. That creates several issues:

1. conflicts with their dependencies prevents upgrade Fybrik third-party components. And prevents resolving security alerts in these third-party components. 
2. The main fybrik file points to these CLI commands only, which is misleading and complicates understanding Fybrik's code.

### PR DESCRIPTION
**Pull request by Vicky-Goldin**
Delete the files that were copied to the `taxonomy-cli` repository.
Update the makefile to use the `taxonomy-cli` commands, and update the site documentation.

Closes #1832 .

### CODE REVIEW

File: manager/apis/app/v1beta1/fybrikapplication_webhook.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 
 	"fybrik.io/fybrik/pkg/environment"
-	validate "fybrik.io/fybrik/pkg/taxonomy/validate"
+	validate "fybrik.io/fybrik/pkg/validate"
```
Comment (roytman) [inline manager/apis/app/v1beta1/fybrikapplication_webhook.go:17]: the import alias is redundant (it has the same name as the package) you can remove it, just leave 
"fybrik.io/fybrik/pkg/validate"

File: manager/apis/app/v1beta1/fybrikmodule_webhook.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 
 	"fybrik.io/fybrik/pkg/environment"
-	validate "fybrik.io/fybrik/pkg/taxonomy/validate"
+	validate "fybrik.io/fybrik/pkg/validate"
```
Comment (roytman) [inline manager/apis/app/v1beta1/fybrikmodule_webhook.go:17]: as above about import

File: site/docs/tasks/custom-taxonomy.md [hunk 3]
```diff
@@ -20,15 +20,15 @@ This task describes how to deploy Fybrik with a custom `taxonomy.json` file that
 
 ## Taxonomy Compile CLI tool 
 
-A CLI tool for compiling a base taxonomy and zero or more taxonomy layers is provided in our repo.
+A CLI tool for compiling a base taxonomy and zero or more taxonomy layers is provided in the [taxonomy-cli repository](https://github.com/fybrik/taxonomy-cli), along with a [Docker image](https://github.com/fybrik/taxonomy-cli/pkgs/container/taxonomy-cli) to directly run the tool.
```
Comment (roytman) [inline site/docs/tasks/custom-taxonomy.md:23]: we don't have to provide a link to th econtaine/taxonomy-cli, you can just mention the docker image.

File: site/docs/tasks/custom-taxonomy.md [hunk 4]
```diff
@@ -20,15 +20,15 @@ This task describes how to deploy Fybrik with a custom `taxonomy.json` file that
...
 
 The base taxonomy can be found in [`charts/fybrik/files/taxonomy/taxonomy.json`](https://github.com/fybrik/fybrik/blob/master/charts/fybrik/files/taxonomy/taxonomy.json) and example layers can be found in [`samples/taxonomy/example`](https://github.com/fybrik/fybrik/tree/master/samples/taxonomy/example).
 
-The following command can be used from the root directory of our repo to run the Taxonomy Compile CLI tool. 
+The following command can be used to run the Taxonomy Compile CLI tool from the provided Docker image.
```
Comment (roytman) [inline site/docs/tasks/custom-taxonomy.md:27]: I'm not sure about "**from** the provided Docker image. I'd say **by** or **with**. (you can validate with  @elsalant )

File: Makefile [hunk 5]
```diff
@@ -32,12 +32,18 @@ all: generate manifests generate-docs verify
 license: $(TOOLBIN)/license_finder
 	$(call license_go,.)
 
+DOCKER_PUBLIC_HOSTNAME ?= ghcr.io
```
[the remark this answers is not in the document]
Comment (Vicky-Goldin) [reply Makefile:35]: I see that the same variables have different values in the makefile and the `docker.mk` file, including the file might interfere with other targets, for example:
`DOCKER_HOSTNAME ?= ghcr.io` in `docker.mk`
`DOCKER_HOSTNAME?=localhost:5000` in `setup-cluster` target in the makefile.
Same for `DOCKER_NAMESPACE`.

Is it intentional, or should I separate the names of the variables? Maybe change the prefix of the variables in `docker.mk` to `DOCKER_MK_<VAR>`?

File: manager/apis/app/v1beta1/fybrikapplication_webhook.go (hunk 1 above)
Comment (Vicky-Goldin) [reply manager/apis/app/v1beta1/fybrikapplication_webhook.go:17]: done

File: manager/apis/app/v1beta1/fybrikmodule_webhook.go (hunk 2 above)
Comment (Vicky-Goldin) [reply manager/apis/app/v1beta1/fybrikmodule_webhook.go:17]: done

File: site/docs/tasks/custom-taxonomy.md (hunk 3 above)
Comment (Vicky-Goldin) [reply site/docs/tasks/custom-taxonomy.md:23]: done

File: site/docs/tasks/custom-taxonomy.md (hunk 4 above)
Comment (Vicky-Goldin) [reply site/docs/tasks/custom-taxonomy.md:27]: Validated

**Comment by roytman** [comment]
The PR https://github.com/fybrik/fybrik/pull/1943 was merged. 
Please add taxonomy-cli version into the requirements.env

**Review by roytman — APPROVED** [review]

**Pull request merged**